src/hooks/useAnnouncementsSettings.tsx

```diff
@@ -0,0 +1,49 @@
+import { useSettings } from 'cozy-client'
+
+const defaultAnnouncements = {
+  dismissedAt: undefined,
+  seen: []
+}
+
+const useAnnouncementsSettings = (): {
+  values?: {
+    dismissedAt?: string
+    seen: string[]
+  }
+  save: (data: { dismissedAt?: string; seen?: string[] }) => void
+} => {
+  const { values, save } = useSettings('home', ['announcements']) as {
+    values?: {
+      announcements: {
+        dismissedAt: string
+        seen: string[]
+      }
+    }
+    save: (data: {
+      announcements: {
+        dismissedAt?: string
+        seen: string[]
+      }
+    }) => void
+  }
```

Comment:
nit: In order to improve readability, we should avoid inlining TS types.

same comment for all other places where types are inlined